### PR TITLE
chore (ai/core): streamText returns result immediately (no Promise)

### PR DESCRIPTION
**Pull request by lgrammel**
## Tasks

- [x] basic implementation
- [x] replace occurrences
- [x] error handling
- [x] 4.0 guide updates
- [x] manual testing
- [x] changeset

### CODE REVIEW

**Review by jeremyphilemon — APPROVED** [review]

File: content/docs/08-troubleshooting/01-migration-guide/25-migration-guide-3-1.mdx
Comment (shaper) [inline content/docs/08-troubleshooting/01-migration-guide/25-migration-guide-3-1.mdx:1]: this edit seems unintentional, as an old migration guide?

Comment (lgrammel) [reply content/docs/08-troubleshooting/01-migration-guide/25-migration-guide-3-1.mdx:1]: good catch

**Comment by patelvivekdev** [comment]
Is there any particular reason for the change from promise to no promise?

**Comment by lgrammel** [comment]
Yes - promises block immediate returns and thus prevent us from streaming back additional data using `StreamData` before the LLM provider responds.

**Pull request merged**